picard/ui/coverartbox.py

```diff
@@ -130,27 +125,88 @@ def set_metadata(self, metadata, item):
             else:
                 # There's no front image, choose the first one available
                 data = metadata.images[0]
-        self.__set_data(data)
-        if item and metadata:
-            self.coverArt.setAcceptDrops(True)
-        else:
-            self.coverArt.setAcceptDrops(False)
+        self.set_data(data)
         release = None
         if metadata:
             release = metadata.get("musicbrainz_albumid", None)
         if release:
-            self.coverArt.setActive(True)
-            self.coverArt.setToolTip(_(u"View release on MusicBrainz"))
+            self.setActive(True)
+            self.setToolTip(_(u"View release on MusicBrainz"))
         else:
-            self.coverArt.setActive(False)
-            self.coverArt.setToolTip("")
+            self.setActive(False)
+            self.setToolTip("")
         self.release = release
 
     def open_release_page(self):
         lookup = self.tagger.get_file_lookup()
         lookup.albumLookup(self.release)
 
     def fetch_remote_image(self, url):
+        return self.parent().fetch_remote_image(url)
+
+
+class CoverArtBox(QtGui.QGroupBox):
+
+    def __init__(self, parent):
+        QtGui.QGroupBox.__init__(self, "")
+        self.layout = QtGui.QVBoxLayout()
+        self.layout.setSpacing(6)
+        self.parent = parent
+        # Kills off any borders
+        self.setStyleSheet('''QGroupBox{background-color:none;border:1px;}''')
+        self.setFlat(True)
+        self.item = None
+        self.cover_art_label = QtGui.QLabel('')
+        self.cover_art_label.setAlignment(QtCore.Qt.AlignTop | QtCore.Qt.AlignHCenter)
+        self.cover_art = CoverArtThumbnail(False, True, parent)
+        spacerItem = QtGui.QSpacerItem(40, 20, QtGui.QSizePolicy.Minimum, QtGui.QSizePolicy.Expanding)
+        self.orig_cover_art_label = QtGui.QLabel('')
+        self.orig_cover_art = CoverArtThumbnail(False, False, parent)
+        self.orig_cover_art_label.setAlignment(QtCore.Qt.AlignTop | QtCore.Qt.AlignHCenter)
+        self.orig_cover_art.setHidden(True)
+        self.view_changes_button = QtGui.QPushButton(_(u'View all changes'), self)
+        self.view_changes_button.setHidden(True)
+        self.layout.addWidget(self.cover_art_label)
+        self.layout.addWidget(self.cover_art)
+        self.layout.addWidget(self.orig_cover_art_label)
+        self.layout.addWidget(self.orig_cover_art)
+        self.layout.addWidget(self.view_changes_button)
+        self.layout.addSpacerItem(spacerItem)
+        self.setLayout(self.layout)
+        self.view_changes_button.clicked.connect(self.show_cover_art_info)
+
+    def show_cover_art_info(self):
+        self.parent.view_info(default_tab=1)
+
+    def _show(self):
+        # We want to show the 2 coverarts only if they are different
+        # and orig_cover_art is not None
+        if getattr(self.orig_cover_art, 'data', None) is None or self.cover_art == self.orig_cover_art:
```

Comment:
orig_cover_art doesn't have a data attribute when the default CD shadow is shown 
![image](https://cloud.githubusercontent.com/assets/16130816/22798657/cee0ed62-ef28-11e6-9dfe-22700f9a6bdc.png)

If I were to remove that check if data is None, the code shows the new coverart image and a blank CD shadow( in this case, IMO it should not show the blank CD shadow indicating that there was no coverart previously)